src/referee.cpp

```diff
@@ -3099,6 +3110,14 @@ HFORef::analyse()
             {
                 M_take_time = 0;
             }
+
+            // Broadcast who has possession. Goalie catches are broadcast separately by the server. Rebroadcasting a message here would overwrite the other one. 
```

Comment:
It's 2 extra ints per cycle, which are negligible. The alternative would be to store the playerOnBall internally in referee, and send it when it changes. But then the parser in agent.cpp needs to handle both cases. I don't think it's worth the complexity.